gravity.sh

```diff
@@ -51,87 +50,99 @@ if [[ -f "${setupVars}" ]];then
+  fi
 
-    # Give time for DNS server to be resolvable
-    secs="30"
-    while [[ "${secs}" -ne 0 ]]; do
-      plural=; [[ "${secs}" -ne 1 ]] && plural="s"
-      echo -ne "${OVER}  ${INFO} Waiting $secs second${plural} before continuing..."
-      sleep 1
-      : $((secs--))
-    done
-
-    # Try again
-    gravity_DNSLookup
-  elif [[ -n "${secs}" ]]; then
-    # Print confirmation of resolvability if it had previously failed
-    echo -e "${OVER}  ${TICK} DNS resolution is now available\\n"
+  # Determine error output message
+  if pidof dnsmasq &> /dev/null; then
+    echo -e "  ${CROSS} DNS resolution is currently unavailable"
+  else
+    echo -e "  ${CROSS} DNS service is not running"
+    "${PIHOLE_COMMAND}" restartdns
   fi
+
+  # Give time for DNS server to be resolvable
+  secs="30"
+  while [[ "${secs}" -ne 0 ]]; do
+    [[ "${secs}" -ne 1 ]] && plural="s"
+    echo -ne "${OVER}  ${INFO} Waiting $secs second${plural} before continuing..."
+    sleep 1
+    : $((secs--))
+  done
+
+  # Try again
+  gravity_DNSLookup
 }
 
 # Retrieve blocklist URLs and parse domains from adlists.list
 gravity_Collapse() {
-  echo -e "  ${INFO} Neutrino emissions detected..."
+  echo -e "  ${INFO} ${COL_BOLD}Neutrino emissions detected${COL_NC}..."
 
-  # Handle "adlists.list" and "adlists.default" files
-  if [[ -f "${adListDefault}" ]] && [[ -f "${adListFile}" ]]; then
+  # Determine if adlists file needs handling
+  if [[ ! -f "${adListFile}" ]]; then
+    # Create "adlists.list" by copying "adlists.default" from internal core repo
+    cp "${adListRepoDefault}" "${adListFile}" 2> /dev/null || \
+      echo -e "  ${CROSS} Unable to copy ${adListFile##*/} from ${piholeRepo}"
+  elif [[ -f "${adListDefault}" ]] && [[ -f "${adListFile}" ]]; then
     # Remove superceded $adListDefault file
     rm "${adListDefault}" 2> /dev/null || \
       echo -e "  ${CROSS} Unable to remove ${adListDefault}"
-  elif [[ ! -f "${adListFile}" ]]; then
-    # Create "adlists.list" by copying "adlists.default" from internal Pi-hole repo
-    cp "${adListRepoDefault}" "${adListFile}" 2> /dev/null || \
-      echo -e "  ${CROSS} Unable to copy ${adListFile##*/} from ${piholeRepo}"
   fi
 
   local str="Pulling blocklist source list into range"
   echo -ne "  ${INFO} ${str}..."
 
   # Retrieve source URLs from $adListFile
-  # Awk Logic: Remove comments (#@;![), CR (windows) line endings and empty lines
-  mapfile -t sources < <(awk '!/^[#@;!\[]/ {gsub(/\r$/, "", $0); if ($1) { print $1 } }' "${adListFile}" 2> /dev/null)
+  mapfile -t sources <<< $(
+    # Logic: Remove comments (#@;![)
+    awk '!/^[#@;!\[]/ {
+      # Remove windows CR line endings
+      gsub(/\r$/, "", $0)
+      # Print non-empty line
+      if ($1) { print $1 }
+    }' "${adListFile}" 2> /dev/null
+  )
 
   # Parse source domains from $sources
-  # Awk Logic: Split by folder/port, remove URL protocol & optional username:password@
-  mapfile -t sourceDomains < <(
+  mapfile -t sourceDomains <<< $(
```

Comment:
this one causing only one adlist (the last one) to be downloaded.

Tested with the following changes locally:
```BASH
#  # Retrieve source URLs from $adListFile
#  mapfile -t sources <<< $(
#    # Logic: Remove comments (#@;![)
#    awk '!/^[#@;!\[]/ {
#      # Remove windows CR line endings
#      gsub(/\r$/, "", $0)
#      # Print non-empty line
#      if ($1) { print $1 }
#    }' "${adListFile}" 2> /dev/null
#  )

mapfile -t sources < <(awk '!/^[#@;!\[]/ {gsub(/\r$/, "", $0); if ($1) { print $1 } }' "${adListFile}" 2> /dev/null)

  # Parse source domains from $sources
#  mapfile -t sourceDomains <<< $(
mapfile -t sourceDomains < <(
    # Logic: Split by folder/port
    awk -F '[/:]' '{
      # Remove URL protocol & optional username:password@
      gsub(/(.*:\/\/|.*:.*@)/, "", $0)
      print $1
    }' <<< "$(printf '%s\n' "${sources[@]}")" 2> /dev/null
  )

  if [[ -n "${sources[*]}" ]] && [[ -n "${sourceDomains[*]}" ]]; then
    echo -e "${OVER}  ${TICK} ${str}"
  else
    echo -e "${OVER}  ${CROSS} ${str}"
    gravity_Cleanup "error"
  fi
}
```